file_store/src/file_sink.rs

```diff
@@ -160,27 +174,36 @@ impl FileSinkClient {
         let bytes = item.encode_to_vec();
         let labels = labels.into_iter().map(Label::from);
 
-        match self.sender.send(Message::Data(on_write_tx, bytes)).await {
-            Ok(_) => {
-                metrics::increment_counter!(
-                    self.metric,
-                    labels
-                        .chain(std::iter::once(OK_LABEL))
-                        .collect::<Vec<Label>>()
-                );
-                tracing::debug!("file_sink write succeeded for {:?}", self.metric);
-                Ok(on_write_rx)
-            }
-            Err(e) => {
-                metrics::increment_counter!(
-                    self.metric,
-                    labels
-                        .chain(std::iter::once(ERROR_LABEL))
-                        .collect::<Vec<Label>>()
-                );
-                tracing::error!("file_sink write failed for {:?} with {e:?}", self.metric);
-                Err(Error::channel())
+        tokio::select! {
+            _ = self.shutdown_listener.clone() => {
+                Err(Error::Shutdown)
```

Comment:
What’s the logic behind labeling a shutdown as an error? Just in the context of a send that won’t complete because of the shutdown ?

Reply:
best i got is `Err(Error::ItIsWhatItIs)` 🤷